src/daft-sql/src/planner.rs

```diff
@@ -913,7 +914,171 @@ impl SQLPlanner {
             SQLExpr::Map(_) => unsupported_sql_err!("MAP"),
             SQLExpr::Subscript { expr, subscript } => self.plan_subscript(expr, subscript.as_ref()),
             SQLExpr::Array(_) => unsupported_sql_err!("ARRAY"),
-            SQLExpr::Interval(_) => unsupported_sql_err!("INTERVAL"),
+            SQLExpr::Interval(interval) => {
+                use regex::Regex;
+
+                /// A private struct represents a single parsed interval unit and its value
+                #[derive(Debug)]
+                struct IntervalPart {
+                    count: i64,
+                    unit: DateTimeField,
+                }
+
+                // Local function to parse interval string to interval parts
+                fn parse_interval_string(expr: &str) -> Result<Vec<IntervalPart>, PlannerError> {
+                    let expr = expr.trim().trim_matches('\'');
+
+                    let re = Regex::new(r"(-?\d+)\s*(year|years|month|months|day|days|hour|hours|minute|minutes|second|seconds|millisecond|milliseconds|microsecond|microseconds|nanosecond|nanoseconds|week|weeks)")
+                        .map_err(|e|PlannerError::invalid_operation(format!("Invalid regex pattern: {}", e)))?;
+
+                    let mut parts = Vec::new();
+
+                    for cap in re.captures_iter(expr) {
+                        let count: i64 = cap[1].parse().map_err(|e| {
+                            PlannerError::invalid_operation(format!("Invalid interval count: {e}"))
+                        })?;
+
+                        let unit = match &cap[2].to_lowercase()[..] {
+                            "year" | "years" => DateTimeField::Year,
+                            "month" | "months" => DateTimeField::Month,
+                            "week" | "weeks" => DateTimeField::Week(None),
+                            "day" | "days" => DateTimeField::Day,
+                            "hour" | "hours" => DateTimeField::Hour,
+                            "minute" | "minutes" => DateTimeField::Minute,
+                            "second" | "seconds" => DateTimeField::Second,
+                            "millisecond" | "milliseconds" => DateTimeField::Millisecond,
+                            "microsecond" | "microseconds" => DateTimeField::Microsecond,
+                            "nanosecond" | "nanoseconds" => DateTimeField::Nanosecond,
+                            _ => {
+                                return Err(PlannerError::invalid_operation(format!(
+                                    "Invalid interval unit: {}",
+                                    &cap[2]
+                                )))
+                            }
+                        };
+
+                        parts.push(IntervalPart { count, unit });
+                    }
```

Comment:
Can we move this into `src/daft-core/src/datatypes/interval.rs` and make this part of the `FromStr` implementation instead.

Reply:
In that case should we also move `daft_sql::error:PlannerError` to a common module? So that won't cause cyclic package dependency when `daft-core` depends on `daft-sql`.

Reply:
hmm, yeah we'll likely have to change the error types. I'm just going to merge this in as is, and when we need the interval fromstr method, we can move it then!.